docs/source/format/Columnar.rst

```diff
@@ -0,0 +1,1266 @@
+     |------------|-------------|-------------|-------------|-------------|--------------|-----------------------|
+     | 0          | 1           | 2           | 1           | 0           | 2            | unspecified (padding) |
+
+    * Children arrays:
+
+      * u0 (Int32):
+        * Length: 6, Null count: 4
+        * Null bitmap buffer:
+
+          |Byte 0 (validity bitmap) | Bytes 1-63            |
+          |-------------------------|-----------------------|
+          |00010001                 | 0 (padding)           |
+
+        * Value buffer:
+
+          |Bytes 0-3   | Bytes 4-7   | Bytes 8-11  | Bytes 12-15 | Bytes 16-19 | Bytes 20-23  | Bytes 24-63           |
+          |------------|-------------|-------------|-------------|-------------|--------------|-----------------------|
+          | 5          | unspecified | unspecified | unspecified | 4           |  unspecified | unspecified (padding) |
+
+      * u1 (float):
+        * Length: 6, Null count: 4
+        * Null bitmap buffer:
+
+          |Byte 0 (validity bitmap) | Bytes 1-63            |
+          |-------------------------|-----------------------|
+          | 00001010                | 0 (padding)           |
+
+        * Value buffer:
+
+          |Bytes 0-3    | Bytes 4-7   | Bytes 8-11  | Bytes 12-15 | Bytes 16-19 | Bytes 20-23  | Bytes 24-63           |
+          |-------------|-------------|-------------|-------------|-------------|--------------|-----------------------|
+          | unspecified |  1.2        | unspecified | 3.4         | unspecified |  unspecified | unspecified (padding) |
+
+      * u2 (`List<char>`)
+        * Length: 6, Null count: 4
+        * Null bitmap buffer:
+
+          | Byte 0 (validity bitmap) | Bytes 1-63            |
+          |--------------------------|-----------------------|
+          | 00100100                 | 0 (padding)           |
+
+        * Offsets buffer (int32)
+
+          | Bytes 0-3  | Bytes 4-7   | Bytes 8-11  | Bytes 12-15 | Bytes 16-19 | Bytes 20-23 | Bytes 24-27 | Bytes 28-63 |
+          |------------|-------------|-------------|-------------|-------------|-------------|-------------|-------------|
+          | 0          | 0           | 0           | 3           | 3           | 3           | 7           | unspecified |
+
+        * Values array (char array):
+          * Length: 7,  Null count: 0
+          * Null bitmap buffer: Not required
+
+            | Bytes 0-6  | Bytes 7-63            |
+            |------------|-----------------------|
+            | joemark    | unspecified (padding) |
+
+Note that nested types in a sparse union must be internally consistent
+(e.g. see the List in the diagram), i.e. random access at any index j
+on any child array will not cause an error.  In other words, the array
+for the nested type must be valid if it is reinterpreted as a
+non-nested array.
```

Comment:
Is that requirement specific for sparse unions? I don't really understand why it's mentioned here.